client/app/utils/routes.js

```diff
@@ -0,0 +1,98 @@
+/* eslint-env commonjs */
+
+import _ from 'lodash';
+
+let Routes = {};
+
+try {
+  Routes = require('../routes/routes.js');
+} catch (e) {
+  console.warn('Can not load route bundle routes.js'); // eslint-disable-line no-console
+}
+
+/**
+ * Examples:
+ *
+ * addDefaultOpts([1, 2, 3], {}) => [1, 2, 3, {}]
+ * addDefaultOpts([1, 2, 3, {format: "json"}], {locale: "en"}) => [1, 2, 3, {format: "json", locale: "en"}]
+ * addDefaultOpts([1, 2, 3, {locale: "fr"}], {locale: "en"}) => [1, 2, 3, {locale: "fr"}]
+ */
+const addDefaultOpts = function addDefaultOpts(args, defaultOpts) {
+  const argsArray = _.toArray(args);
+  const last = _.last(argsArray);
+
+  if (last && _.isObject(last)) {
+    return _.initial(argsArray).concat([_.assign({}, defaultOpts, last)]);
+  } else {
+    return argsArray.concat([defaultOpts]);
+  }
+};
+
+const didYouMean = function didYouMean(routeName) {
+  if (_.endsWith(routeName, '_path') && Routes[routeName]) {
+    return `Did you mean '${routeName.replace(/_path$/, '')}'?`;
+  } else {
+    return null;
+  }
+};
+
+const routeNameToPathHelperName = function routeNameToPathHelperName(routeName) {
+  return `${routeName}_path`;
+};
+
+const wrapWithDefaultOpts = function wrapWithDefaultOpts(pathFns, defaultOpts) {
+  return pathFns.reduce((routeObject, { pathHelperName, pathHelper }) => {
+    const withDefaultOptsFn = function withDefaultOpts(...args) {
+      return pathHelper(...addDefaultOpts(args, defaultOpts));
+    };
+
+    // Copy the toString function.
+    // It contains the path spec, which might be useful
+    // For example:
+    // single_conversation_path.toString => (/:locale)/:person_id/messages/:conversation_type/:id(.:format)
+    withDefaultOptsFn.toString = pathHelper.toString;
+
+    routeObject[pathHelperName] = withDefaultOptsFn; // eslint-disable-line no-param-reassign
+    return routeObject;
+  }, {});
+};
+
+//
+// Public API
+//
+
+/**
+ * Creates a subset of all routes.
+ *
+ * You can pass also `defaultOpts` object, for example for "locale"
```

Comment:
I find the name defaultOpts a bit misleading. If I get this right these are global arguments that are passed to all reverse routing functions so you don't have to pass them separately in every call? If so then I suggest renaming to globalParams.